src/main/java/roomescape/dto/reservation/MyReservationResponseDto.java

```diff
@@ -0,0 +1,26 @@
+package roomescape.dto.reservation;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import roomescape.domain.Reservation;
+
+public record MyReservationResponseDto(
+        long reservationId,
+        String theme,
+        LocalDate date,
+        @JsonFormat(pattern = "HH:mm")
+        LocalTime time,
+        String status
+) {
+
+    public MyReservationResponseDto(Reservation reservation) {
+        this(
+                reservation.getId(),
+                reservation.getTheme().getName(),
+                reservation.getDate(),
+                reservation.getTime().getStartAt(),
```

Comment:
```suggestion
                reservation.getStartAt(),
```
단일 호출 메서드를 추가하면 바로 사용할 수 있을 것 같아요.

Reply:
오호...! 이 방법은 생각하지 못했는데요, 그 이유가 무엇인지 궁금합니다!

이 방법에 대한 저의 고민은 아래와 같은데요,
1. reservationResponseDto에서 호출하는 getter를 모두 단일 호출 메서드로 바꾼다면 **getter가 늘어납니다**. 
  가령 현재 MyReservationResponseDto에서 제공하는 메서드를 모두 단일 호출로 바꾼다면, Reservation 객체에는 `getThemeName`, `getStartAt`라는 메서드가 두가지가 생깁니다. 

2. **도메인과 DTO의 경계가 흐려집니다.** 도메인에서 응답 형식에 맞는 메서드를 모두 만든다면, 응답 형식이 변경되면 도메인이 함께 변경될 수 있고, 응답 dto가 가진 역할이 흐려집니다.

이에 대한 제로의 의견이 궁금합니다. 🤔 

아니면 혹시 Time의 경우엔 startAt 필드만 있으므로, 단일호출 메서드로 예외적으로 변경하는 것을 권장하시는거라면, **응답 dto에 포함해야 하는 Time의 필드가 (startAt 외에도) 더 늘어날 경우는 고려하지 않아도 될지,** 질문드립니다!!

Reply:
이렇게 제안드린 이유는, 현재 구조상 DTO가 도메인을 깊게 탐색하며 강하게 의존하고 있기 때문입니다.
사실 DTO가 도메인 자체를 직접 의존하고 있는 구조에 대해서 먼저 리뷰를 드리고 싶었지만,
이번 미션의 요구사항 및 PR의 직접적인 변경사항이 아니었어서 우선순위에서는 다소 미뤄두었던 부분이에요.

reservation.getTime().getStartAt()처럼 DTO가 도메인의 내부 필드 구조까지 깊게 접근하는 방식은 캡슐화가 많이 깨지는 구조라 생각돼요.
이런 경우, 도메인 내부 구조에 변경이 생기면 DTO까지 영향이 퍼질 수 있어서, 의존도를 조금이라도 낮춰보고자 제안드렸습니다.

꼭 메서드를 추가해야 한다는 뜻이라기보다는, 도메인의 표현을 외부에서 조합하기보단, 도메인이 의미를 응집해서 표현해주는 방향이 더 객체지향적인 설계일 수 있다는 관점에서 드렸던 의견입니다.

추가로 참고하면 좋을자료 공유드립니다.
- https://umbum.dev/1206/
- https://tecoble.techcourse.co.kr/post/2021-04-25-dto-layer-scope/